content/en/blog/low-hanging-performance/index.md

```diff
@@ -0,0 +1,460 @@
+   collection because it's such a significant topic that even one book is insufficient.)
+1. **A function call is computationally expensive** if the compiler cannot [inline
+   expanse](https://en.wikipedia.org/wiki/Inline_expansion) it.
+1. **Minimize the problem space at every level of abstraction** and the need
+   for variables, i.e. especially in inner loops. Consider what parts of
+   inputs are really varying and what parts are constant. For example, think
+   twice if you need regexp inside of your program.
+
+## Code Readability
+
+We want to maximize our **code’s readability**. One of the Go code’s problems is
+that it overuses the if-statement, which prevents you from noticing the
+algorithm's critical decision points.
+
+For example, Go’s standard library includes quite many of the following code
+blocks:
+
+```go
+func doSomething(p any, b []byte) {
+    if p == nil {
+        panic("input argument p cannot be nil")
+    }
+    if len(b) == 0 {
+        panic("input argument p cannot be nil")
+    }
+    ...
+    err := w.Close()
+    if err != nil {
+        log.Fatal(err)
+    }
+}
+```
+
+It’s easy to see that together with Go’s if-based error checking, these two hide
+the happy path and make it difficult to follow the algorithm and skim the code.
+The same thing can be found from Go’s unit tests if no 3rd party helper package
+is used:
+
+```go
+     for _, tt := range tests {
+          t.Run(tt.name, func(t *testing.T) {
+               r := <-Open(tt.args.config, tt.args.credentials)
+               if got := r.Err(); !reflect.DeepEqual(got, tt.want) {
+                    t.Errorf("Open() = %v, want %v", got, tt.want)
+               }
+               w := r.Handle()
+               if got := (<-Export(w, tt.args.exportCfg)).Err(); !reflect.DeepEqual(got, tt.want) {
+                    t.Errorf("Export() = %v, want %v", got, tt.want)
+               }
+               if got := (<-Close(w)).Err(); !reflect.DeepEqual(got, tt.want) {
+                    t.Errorf("Close() = %v, want %v", got, tt.want)
+               }
+          })
+     }
+```
+
+The above code block is from a different test than the block below, but I think
+you get the idea. I’m speaking fast skimming of code where simplicity and form
+help a lot. Very much similar to syntax highlighting.
+
+```go
+func TestNewTimeFieldRFC3339(t *testing.T) {
+     defer assert.PushTester(t)()
+
+     var testMsg Basicmessage
+     dto.FromJSON([]byte(timeJSONRFC3339), &testMsg)
+     timeValue := testMsg.SentTime
+
+     assert.INotNil(timeValue)
+     assert.Equal(timeValue.Year(), 2022)
+     assert.Equal(timeValue.Month(), time.September)
+     assert.Equal(timeValue.Day(), 30)
+}
+```
+
+That's clear and easy to skim. It's also straightforward to extend to use table
+testing. I left it for the reader to figure out how that code renders without
+an assert package.
+
+## Function Inline Expansion
```

Comment:
Inline expansion is not a familiar term for all, and most of the people will not open the wikipedia link. How about explaining it here using hard wire 😄